### PR TITLE
Use Clang as the compiler for ConvPerf

### PR DESCRIPTION
**Pull request by mariecwhite**
ConvPerf fails intermittently with a "corrupted double-linked list" error. Explicitly set compiler to Clang instead of gcc.

### CODE REVIEW

File: build_tools/benchmarks/convperf/build_and_run_convperf.sh [hunk 1]
```diff
@@ -47,7 +47,10 @@ popd # external/iree
 popd # ${REPO_DIR}
 
 # Build ConvPerf.
-cmake -GNinja -B "${BUILD_DIR}" "${REPO_DIR}"
+cmake -GNinja \
+    -DCMAKE_CXX_COMPILER=/usr/bin/clang++ \
+    -DCMAKE_C_COMPILER=/usr/bin/clang \
```
Comment (GMNGeoffrey) [inline build_tools/benchmarks/convperf/build_and_run_convperf.sh:52]: This hardcodes a specific path at which to find clang, which isn't great. Better would be `"$(which clang)"` or I think just setting it as "clang" would make CMake do the same lookup. Generally though, we encode this in `CC` and `CXX` env variables. Those are set in the Docker containers or, e.g. https://github.com/iree-org/iree/blob/ceb9cfd8c955f8fed500112702c7f7cafc2fc38d/build_tools/docker/dockerfiles/base.Dockerfile#L34 or the command line when they need to vary e.g. https://github.com/iree-org/iree/blob/ceb9cfd8c955f8fed500112702c7f7cafc2fc38d/.github/workflows/ci.yml#L677

Comment (mariecwhite) [reply build_tools/benchmarks/convperf/build_and_run_convperf.sh]: I added environment variables to the DockerFile.

File: build_tools/benchmarks/convperf/build_and_run_convperf.sh [hunk 2]
```diff
@@ -47,7 +47,10 @@ popd # external/iree
 popd # ${REPO_DIR}
 
 # Build ConvPerf.
-cmake -GNinja -B "${BUILD_DIR}" "${REPO_DIR}"
+cmake -GNinja \
+  -DCMAKE_C_COMPILER="${CC:-clang}" \
+  -DCMAKE_CXX_COMPILER="${CXX:-clang++}" \
```
Comment (GMNGeoffrey) [inline build_tools/benchmarks/convperf/build_and_run_convperf.sh:52]: At the point you've set CC and CXX, it should be necessary to explicitly specify these arguments here, but it's OK if you want to.

File: .github/workflows/run_convperf.yml [hunk 3]
```diff
@@ -17,6 +17,7 @@ on:
   schedule:
     - cron: '0 16 * * *'
   workflow_dispatch:
+  pull_request:
```
Comment (GMNGeoffrey) [inline .github/workflows/run_convperf.yml:20]: Reminder to revert

**Review by GMNGeoffrey — APPROVED** [review]

File: build_tools/benchmarks/convperf/build_and_run_convperf.sh (hunk 2 above)
Comment (mariecwhite) [reply build_tools/benchmarks/convperf/build_and_run_convperf.sh:52]: I'll leave it in :)

File: .github/workflows/run_convperf.yml (hunk 3 above)
Comment (mariecwhite) [reply .github/workflows/run_convperf.yml]: Removed

**Pull request merged**